WriteGCode.py

```diff
@@ -149,7 +151,10 @@ def write_gcode(config, commands) -> typing.Tuple[str, cura.LayerDataBuilder.Lay
 
 				gcode = ""
 				if not is_retracted and retraction_enable:
-					gcode += "G0 F{speed} E{e:.6f}\n".format(speed=retraction_speed * 60, e=e - retraction_distance)
+					gcode += "G0 F{speed} E{e:.6f}".format(speed=retraction_speed * 60, e=e - retraction_distance)
+					if retraction_hop_enabled:
+						gcode += " Z{z:.6f}".format(z=layer_heights[layer_nr] + retraction_hop)
+					gcode += " (RETRACT)\n"
```

Comment:
I think this is your debug code? Or maybe something specific to your device? Having the word `(RETRACT)` there is not [standard g-code](https://reprap.org/wiki/G-code#G0_.26_G1:_Move).

Reply:
```suggestion
					gcode += "G10\n" #enable retraction gcode
```

This seems to work for me